packages/connect-sql/src/data-type.ts

```diff
@@ -0,0 +1,81 @@
+import { Database, RunResult, Statement } from "better-sqlite3";
+import { DBConnection, Store, UploadDataFnParams } from "./types";
+import { SQLiteConnection } from "./sqlite-adapter";
+
+export interface DataType {
+    readonly type: 'data'
+    readonly car: string
+    readonly size: number
+    readonly data: Uint8Array
+}
+
+export interface DataRecord {
+    readonly car: string;
+    readonly blob: Uint8Array
+    readonly created_at: Date;
+}
+
+export class DataSQLRecordBuilder {
+    #car?: string;
+    #size?: string;
+    #data?: Uint8Array;
+
+    static fromUploadParams(data: Uint8Array, params: UploadDataFnParams): DataSQLRecordBuilder {
+        const bld = new DataSQLRecordBuilder()
+        bld.#car = params.car
+        bld.#data = data
+        bld.#size = params.size
```

Comment:
I think we don't use`size` anywhere so perhaps worth removing if that's true. It's purpose is ostensibly as a gaurd against the s3 signed upload URL handler accepting DOS sized files, but I don't see it getting used there. Maybe worth adding there if possible, otherwise worth removing from the entire `UploadDataFnParams` type across connectors. This is a good time in the release lifecycle to do these kinds of global changes.

Reply:
https://github.com/fireproof-storage/valid-cid-s3-bucket/blob/0f8f8d44f11cd4bd8b7665316d6676b49014830e/getSignedURL/app.ts#L220

Reply:
i think we should not mix up things here --- this is about to create a sql adapter --> and your approach is totally fine but should be done in a future PR.